### PR TITLE
Set cache control headers in V2 API to mirror V1

### PR DESCRIPTION
**Pull request by alexweav**
The V1 API sends CORS headers + the `Cache-Control` header, all the time:
https://github.com/prometheus/alertmanager/blob/87ad8437fc1e28280e8c5c5fdcb41e0a8904a855/api/v1/api.go#L44

In V2, we replaced this with [rs/cors](https://github.com/rs/cors). But, the `Cache-Control` header is not actually a CORS-related header, so that library does not manage it. This means the `Cache-Control` headers are missing in V2.

This PR adds the header in the V2 API, matching V1.

Fixes https://github.com/prometheus/alertmanager/issues/3175

Signed-off-by: Alex Weaver <weaver.alex.d@gmail.com>

### CODE REVIEW

File: api/v2/api.go [hunk 1]
```diff
@@ -130,11 +130,24 @@ func NewAPI(
...
 
 	handleCORS := cors.Default().Handler
-	api.Handler = handleCORS(openAPI.Serve(nil))
+	api.Handler = handleCORS(setResponseHeaders(openAPI.Serve(nil)))
 
 	return &api, nil
 }
 
+var responseHeaders = map[string]string{
+	"Cache-Control": "no-cache, no-store, must-revalidate",
```
Comment (alexweav) [inline api/v2/api.go:139]: The specific headers I chose exactly match what V1 sends. Though, perhaps now is the time to revisit this?

Specifically, `no-store` is stronger than the other two headers and disables caching completely **in later specifications.** In older specifications, it made sense to use both `no-cache` and `no-store` at the same time. So, I kept both. See: https://stackoverflow.com/questions/866822/why-both-no-cache-and-no-store-should-be-used-in-http-response

Comment (gotjosh) [reply api/v2/api.go:139]: I'd say we drop it and keep only `no-store` the specification of `no-store` is quite old so it seems sensible to rely on clients having implemented that already.

Comment (alexweav) [reply api/v2/api.go]: Done!

**Comment by gotjosh** [comment]
Thank you very much for your contribution.

**Pull request merged**